platform-sdk/swirlds-benchmarks/build.gradle.kts

```
@@ -45,3 +49,18 @@ jmh {
 }
 
 fun listProperty(value: String) = objects.listProperty<String>().value(listOf(value))
+
+tasks.register("jmhReconnect") {
+    // Unfortunately, this is the only way to re-configure the jmh extension/plugin today,
+    // and one can only ever run a single jmh task in a given build run.
+    // So either run `./gradlew jmh` to run the regular JMH task defined above,
+    // or run `./gradlew jmhReconnect` to run the override below:
+    doFirst {
+        jmh.includes.set(listOf("Reconnect.*"))
+        jmh.jvmArgs.set(listOf("-Xmx16g"))
+        jmh.fork.set(1)
+        jmh.warmupIterations.set(2)
+        jmh.iterations.set(5)
+    }
+    finalizedBy("jmh")
+}
```

This is probably just not well documented. You can register additional `JMHTask` instances and have a cleaner configuration:

```suggestion
tasks.register<JMHTask>("jmhReconnect") {
    group = "jmh"
    jarArchive.set(tasks.jmhJar.flatMap { it.archiveFile })
    
    includes.set(listOf("Reconnect.*"))
    jvmArgs.set(listOf("-Xmx16g"))
    fork.set(1)
    warmupIterations.set(2)
    iterations.set(5)
    
    resultsFile.convention(layout.buildDirectory.file("results/jmh/results-reconnect.txt"))
}
```

Note that you still need to run JMH tests with `--no-configuration-cache` until the next release of the JMH plugin which will fix that issue.

---

Thanks for the suggestion! In fact, I started implementing this using a JMHTask, but I got stuck with the `group` and `jarArchive` definitions, and then I found the `finalizedBy` solution specifically for jmh and went with it.

However, your code looks a bit cleaner than the previous solution, so I've applied this suggestion. Note that I had to also specify the `jvm` parameter for jmh, otherwise it defaulted to an older JVM version that happens to be installed on my computer, and then the benchmark wouldn't run. With the new definition, it runs just fine.

As for the `--no-configuration-cache`, I found that the hard way and I simply set `org.gradle.configuration-cache=false` in my `~/.gradle/gradle.properties`. I know this might be inefficient. However, I prefer when things work rather than fail because of an "optimization". :)

---

Thanks. I missed the `jvm` part. We could move `group`, `jarArchive`, `jvm` to `com.hedera.hashgraph.benchmark-conventions.gradle.kts` as follows. Then we have this configured centrally for this and all future JMHTasks.

```
tasks.withType<JMHTask>().configureEach {
    group = "jmh"
    jarArchive = tasks.jmhJar.flatMap { it.archiveFile }
    jvm = javaToolchains.launcherFor(java.toolchain).map { it.executablePath }.get().asFile.path
}

```

---

Good idea. I moved these parameters to `com.hedera.hashgraph.benchmark-conventions.gradle.kts`.